cmd/admin-bucket-quota.go

```diff
@@ -0,0 +1,159 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/console"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var adminQuotaFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "fifo",
+		Usage: "Set fifo quota,allowing automatic deletion of older content",
+	},
+	cli.BoolFlag{
+		Name:  "hard",
+		Usage: "Set a hard quota, disallowing writes after quota is reached",
+	},
+}
+
+// quotaMessage container for content message structure
+type quotaMessage struct {
+	op        string
+	Status    string `json:"status"`
+	Bucket    string `json:"bucket"`
+	Quota     uint64 `json:"quota,omitempty"`
+	QuotaType string `json:"type,omitempty"`
+}
+
+func (q quotaMessage) String() string {
+	if q.op == "set" {
+		return console.Colorize("QuotaMessage",
+			fmt.Sprintf("Successfully set bucket quota of %s with %s type on `%s`", humanize.IBytes(q.Quota), q.QuotaType, q.Bucket))
+	}
+	return console.Colorize("QuotaInfo",
+		fmt.Sprintf("Bucket `%s` has %s quota of %s", q.Bucket, q.QuotaType, humanize.IBytes(q.Quota)))
+}
+
+func (q quotaMessage) JSON() string {
+	q.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(q, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+var adminBucketQuotaCmd = cli.Command{
+	Name:   "quota",
+	Usage:  "Set/Get bucket quota",
+	Action: mainAdminBucketQuota,
+	Before: setGlobalsFromContext,
+	Flags:  append(adminQuotaFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+ 
+USAGE:
+   {{.HelpName}} TARGET USERNAME
+ 
+UNITS
+   quota accepts human-readable case-insensitive number
+   suffixes such as "k", "m", "g" and "t" referring to the metric units KB,
+   MB, GB and TB respectively. Adding an "i" to these prefixes, uses the IEC
+   units, so that "gi" refers to "gibibyte" or "GiB". A "b" at the end is
+   also accepted. Without suffixes the unit is bytes.
+ 
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
+   1. Display bucket quota configured for "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket
+	
+   2. Set FIFO quota for a bucket "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket 64kB --fifo
+`,
+}
+
+// checkAdminBucketQuotaSyntax - validate all the passed arguments
+func checkAdminBucketQuotaSyntax(ctx *cli.Context) {
```

Comment:
validation should fail if 
- [ ] a bucket wasn't provided
e.g `./mc admin bucket quota play`

- [ ] if we supply additional positional arguments without `--hard` or `--fifo` flags 
e.g `./mc admin bucket quota something`